### ISSUE TITLE
Exclude SSID from writing to file  - Enhancement

### ISSUE THREAD

**Issue by gjchester** [JosephHewitt/wardriver_rev3#6]
My dashcam (Road Angel Halo ) generates wifi hotspot to allow you to download data, the hotspot is active all the time the camera is running, so in effect any time the car is running.  

Would you consider adding an exclude list where you can enter a few (maybe even just one or two) SSIDs and they are ignored so are not written to the SD card or show in the detected nubmers, My Wardriver never shows zero as even in emptt areas its picking up the dashcam.

I can also see it being an advantage for peeple who have cars that generate a passenger use hotspot (Audi, Tesla, and BMW offer it as a subscription I think),  or have a phone in tether mode to provide data in car (say to an Alexa or Chromecast) to play music in car.

**Comment by JosephHewitt** [comment]
Hi,

Yes, this seems like something a few people would find useful. It's not something I want to work on right now but have no objections to anybody else working on it. I'll keep the issue open and tag it with enhancement so maybe someone will pick it up before I do.

Thanks for the suggestion.

**Comment by grymoire** [comment]
I'd use this to exclude my home or work SSID  from wigle. In the meantime I'll write a shell script to edit the CSV files.
It might even be nice to have a way to block out a certain physical area (like a city block), like an eraser.

**Comment by JosephHewitt** [comment]
I gave this some more thought and realised it is probably something which lots of people would find useful for various reasons. I'll assign this issue to myself and work on it somewhat soon.

My intention is to create a text box on the setup page somewhere in which you can enter multiple SSIDs and they will all be ignored from the scanning.

In regards to blanking out a whole area, I think this could cause suspicious blank areas on the Wigle maps which may actually stand out more than just submitting your data as normal. I'll have to think about this some more, I'm not sure yet if it's something many people would find useful.

**Comment by gjchester** [comment]
I would have thought that trying to block an area isn’t really viable.   Remember your only stopping your own hardware picking things up in a set block, whereas anyone else who comes past with a wardriving kit would detect your SSID’s.

My thoughts were purely to see if it was viable to exclude (for example) a WI-FI hotspot of HaloPro_1234 (yes I did change the password from the defaults) with the same MAC address, well BSSID, being detected in lots of locations all over the UK depending where the car is when I start the engine as that will start both the dash cam and the wardriver hardware.

**Comment by JosephHewitt** [comment]
I can understand why someone would block a whole area. For example, if you work in a strict area you may have a contract stating you can't release any information about the infrastructure of the building; wardriving there would probably breach such a contract. It is true that other people could submit that data instead but this feature would still protect the user from legal issues in some situations. Therefore I do generally like this feature but I don't consider it as important as @gjchester 's initial issue (blocking specific SSIDs/BSSIDs).

I have noticed that it is possible to track cars based on their WiFi networks or dashcam networks so I will indeed add this feature to protect the privacy of anybody wardriving with such devices/vehicles.

In the meantime, I suggest contacting Wigle and requesting that they remove your dashcam BSSID. Since other people may detect your dashcam too, this is a more ideal solution to the problem anyway.

**Comment by grymoire** [comment]
Yes. I was thinking of the case where someone uploads data to wigle, and the CEO is told their network data was uploaded by an employee. Yes, it's public info. No, the CEO doesn't care.

Also - some people live in remote locations, and may prefer to not increase their attack surface.

I plan to write a shell/awk/sed script that removes devices, btw.

And true - I don't want the Bluetooth of my car+phone mapped in wigle - it would be able to track every route I took to gather information.

**Comment by grymoire** [comment]
ALso - I noticed my home WiFi was listed as [OPEN], I assumed it was because I gave it my PSK.

**Comment by grymoire** [comment]
I've created some Linux shell scripts that can filter out MAC addresses,SSID's,and devices that are nearby. Let me know if anyone would like them.

**Comment by JosephHewitt** [comment]
> I assumed it was because I gave it my PSK.

This should never be the case. The scan results are unaffected by your network configuration so this sounds like a bug.

Could you please open a new issue and include the expected security type of your network? Thank you.

**Comment by grymoire** [comment]
I created a fork with some scripts to exclude devices by location - SSID or MAC address. in https://github.com/grymoire/wardriver_rev3

**Comment by signout** [comment]
> I gave this some more thought and realised it is probably something which lots of people would find useful for various reasons. I'll assign this issue to myself and work on it somewhat soon.
Thank you :)


> My intention is to create a text box on the setup page somewhere in which you can enter multiple SSIDs and they will all be ignored from the scanning.
Please extend this to MAC addresses as well. I'd like to be able to just feed it a list of both my wifi and bluetooth addresses and have them excluded no matter what.
It's also entirely likely that someone else has the same (funny, probably found on a "funniest wifi names"-list on the interwebs) SSID :)

**Comment by grymoire** [comment]
The SD-card implementation sounds like a good idea. 

BTW, in my patches (I didn't issue a pull request) I have a make file and all I do is mount the SD card, cd to the wardriver directory, type "make", then to to wifle.net, and upload one GZIP file called "a.tar.zip" that contains all of the recent (since last upload) files.  It sounds compatible with your design. I also have a blacklist by geolocation, in my make file script built in. It has a central location and a +/- distance around the location. This could also be on the SD card.

Since you are looking an improvements, it would be nice to have the default starting SSID kept on the SD card. I lost a lot of potential captures when I reflashed the firmware a second time , and drove for days without doing the first time setup  (setting the SSID and password). This could be on the SD card once learned.

**Comment by JosephHewitt** [comment]
> BTW, in my patches (I didn't issue a pull request) I have a make file and all I do is mount the SD card, cd to the wardriver directory, type "make", then to to wifle.net, and upload one GZIP file called "a.tar.zip" that contains all of the recent (since last upload) files. It sounds compatible with your design. I also have a blacklist by geolocation, in my make file script built in. It has a central location and a +/- distance around the location. This could also be on the SD card.

Good idea. However, this isn't something I want to integrate with this project. I want the wardriver to have as much functionality built in as viable and would prefer not to require a PC for some features. 

> Since you are looking an improvements, it would be nice to have the default starting SSID kept on the SD card. I lost a lot of potential captures when I reflashed the firmware a second time , and drove for days without doing the first time setup (setting the SSID and password). This could be on the SD card once learned.

Thanks for the suggestion. I created issue #32 for this.

**Comment by JosephHewitt** [comment]
This feature is now implemented and merged into `main` with PR #29 

As always, code in `main` works but may not be as stable as a full release. I suggest waiting for the release before flashing your wardrivers with this code.

To use this feature, create a file called `bl.txt` on the SD card of your wardriver. This file can contain up to 20 lines*, each is a MAC address or SSID you want to ignore. It is case sensitive and MAC addresses should be uppercase only.

*this limit is mostly arbitrary for now and may be changed once more testing is complete.
